src/jit/flowgraph.cpp

```diff
@@ -21721,6 +21742,58 @@ void Compiler::fgDebugCheckBlockLinks()
     }
 }
 
+//------------------------------------------------------------------------------
+// fgDebugCheckNodesUniqueness: Check that each tree in the method has its own unique gtTreeId.
+//
+void Compiler::fgDebugCheckNodesUniqueness()
+{
+    unsigned     numberOfTrees = JitTls::GetCompiler()->compGenTreeID;
+    BitVecTraits nodesVecTraits(numberOfTrees, this);
+    BitVec       uniqueNodes(BitVecOps::MakeEmpty(&nodesVecTraits));
+
+    ArrayStack<unsigned> stack(this);
+
+    for (BasicBlock* block = fgFirstBB; block; block = block->bbNext)
+    {
+        if (block->IsLIR())
+        {
+            for (auto i : LIR::AsRange(block))
+            {
+                stack.Push(i->gtTreeID);
+            }
+        }
+        else
+        {
+            for (GenTreeStmt* stmt = block->firstStmt(); stmt; stmt = stmt->gtNextStmt)
+            {
+                GenTreePtr root = stmt->gtStmtExpr;
+                fgWalkTreePre(&root, fgMarkTreeId, &stack);
+            }
+        }
+
+        while (stack.Height() != 0)
+        {
+            unsigned gtTreeID = stack.Pop();
+            assert(!BitVecOps::IsMember(&nodesVecTraits, uniqueNodes, gtTreeID));
+            BitVecOps::AddElemD(&nodesVecTraits, uniqueNodes, gtTreeID);
+        }
+    }
+}
+
+//------------------------------------------------------------------------
+// fgMarkTreeId: Visit all subtrees in the tree and memorize gtTreeIDs.
+//
+// Arguments:
+//    pTree            - Pointer to the tree to walk
+//    fgWalkPre        - ArrayStack to memorize gtTreeID
+//
+Compiler::fgWalkResult Compiler::fgMarkTreeId(GenTree** pTree, fgWalkData* fgWalkPre)
+{
+    ArrayStack<unsigned>* stack = reinterpret_cast<ArrayStack<unsigned>*>(fgWalkPre->pCallbackData);
```

Comment:
`pCallbackData` is `void*` so `static_cast` is enough

Reply:
Fixed, I will create a PR to fix other places where we use `reinterpret_cast` in the walkers.